src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java

```diff
@@ -144,23 +155,35 @@ public void finishLocalSetup() {
         childContext = checkContext;
     }
 
+    /**
+     * {@inheritDoc} Creates child module.
+     * @noinspection ChainOfInstanceofChecks
```

Comment:
Your example from your forked repo, inside main repo this suppression exists
https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/Checker.java#L442

Otherwise TeamCity fails